windows/win_nssm.ps1

```diff
@@ -0,0 +1,547 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2015, George Frank <george@georgefrank.net>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+$ErrorActionPreference = "Stop"
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$params = Parse-Args $args;
+$result = New-Object PSObject;
+Set-Attr $result "changed" $false;
+
+
+If ($params.name)
+{
+    $name = $params.name
+}
+Else
+{
+    Fail-Json $result "missing required argument: name"
+}
+
+If ($params.state)
+{
+    $state = $params.state.ToString().ToLower()
+    $validStates = "present", "absent", "started", "stopped", "restarted"
+    
+    # These don't really fit the declarative style of ansible
+    # If you need to do these things, you can just write a command for it
+    # "paused", "continued", "rotated"
+
+    If ($validStates -notcontains $state)
+    {
+        Fail-Json $result "state is $state; must be one of: $validStates"
+    }
+}
+else
+{
+    $state = "present"
+}
+
+If ($params.application)
+{
+    $application = $params.application
+}
+Else
+{
+    $application = $null
+}
+
+If ($params.app_parameters)
+{
+    $appParameters = $params.app_parameters
+}
+Else
+{
+    $appParameters = $null
+}
+
+If ($params.stdout_file)
+{
+    $stdoutFile = $params.stdout_file
+}
+Else
+{
+    $stdoutFile = $null
+}
+
+If ($params.stderr_file)
+{
+    $stderrFile = $params.stderr_file
+}
+Else
+{
+    $stderrFile = $null
+}
+
+Function Service-Exists
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name
+    )
+
+    return ,[bool](Get-Service "$name" -ErrorAction SilentlyContinue)
```

Comment:
This one isn't needed, so I'll remove it (it was just habit!). The leading comma does, however, have a use when multiple items are returned (more info: http://stackoverflow.com/questions/25279112/returning-a-liststring-using-a-comma-vs-not-using-a-comma)